pkg/genericcomp/genericcomp.go

```diff
@@ -0,0 +1,162 @@
+// Copyright (c) 2020, Oracle and/or its affiliates.
+// Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
+
+package genericcomp
+
+import (
+	v1beta1v8o "github.com/verrazzano/verrazzano-crd-generator/pkg/apis/verrazzano/v1beta1"
+	"github.com/verrazzano/verrazzano-operator/pkg/types"
+	"github.com/verrazzano/verrazzano-operator/pkg/util"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// GenericComponentSelectorLabel defines the selector label for generic component resources.
+const GenericComponentSelectorLabel = "verrazzano.name"
+
+// NewDeployment constructs a deployment for a generic component.
+func NewDeployment(generic v1beta1v8o.VerrazzanoGenericComponent, namespace string, labels map[string]string) *appsv1.Deployment {
+	return &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      generic.Name,
+			Namespace: namespace,
+			Labels:    labels,
+		},
+		Spec: appsv1.DeploymentSpec{
+			Replicas: func() *int32 {
+				if generic.Replicas == nil {
+					return util.NewVal(1)
+				}
+				return generic.Replicas
+			}(),
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					GenericComponentSelectorLabel: generic.Name,
+				},
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
```

Comment:
Do we support letting user add other labels here?

Reply:
No, originally that was the plan but Mike Gilbode said that was not needed for the initial deliverable

Reply:
The initial CRD we put together allowed for user specified 
labels and annotations to be set in the generic component definition in the model.